File: ToDoModel/Data Model/Project/ProjectList.vb


<Serializable()>
Public Class ProjectList

    Public ProjectDictionary As Dictionary(Of String, String)

    Public Sub New(ByVal dictProjectList As Dictionary(Of String, String))
        ProjectDictionary = dictProjectList
    End Sub

    Public Sub ToCSV(FileName As String)
        Dim csv As String = String.Join(
            Environment.NewLine,
            ProjectDictionary.[Select](Function(d) $"{d.Key};{d.Value};"))
        System.IO.File.WriteAllText(FileName, csv)
    End Sub

End Class

File: ToDoModel/Data Model/Project/ProjectList.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;

namespace ToDoModel
{
    [Serializable()]
    public class ProjectList
    {

        public Dictionary<string, string> ProjectDictionary;

        public ProjectList(Dictionary<string, string> dictProjectList)
        {
            ProjectDictionary = dictProjectList;
        }

        public void ToCSV(string FileName)
        {
            string csv = string.Join(Environment.NewLine, ProjectDictionary.Select(d => $"{d.Key};{d.Value};"));
            System.IO.File.WriteAllText(FileName, csv);
        }

    }
}